src/main/kotlin/http/KtorServer.kt

```diff
@@ -55,15 +46,18 @@ private fun createIncomingRequest(req: ApplicationRequest) =
         req.headers.toMap(),
         req.queryString(),
         req.host(),
-        req.headerInt("Content-Length").let { len ->
-            if (len == 0) {
-                RETURN_EMPTY_BUF
-            } else {
-                { req.call.receiveStream().readBytes(len) }
-            }
-        }
+        getReadBytesFn(req),
+        getReadBytesFn(req)
     )
 
+fun getReadBytesFn(req: ApplicationRequest) =
+    req.headerInt("Content-Length").let { len ->
+    if (len == 0) {
+        RETURN_EMPTY_BUF
+    } else {
+        { req.call.receiveStream().readBytes(len) }
```

Comment:
Does this allow reading the request twice?

Reply:
No, next commit fixes it.